astropy/coordinates/representation/geodetic.py

```diff
@@ -10,6 +11,16 @@
 from .cartesian import CartesianRepresentation
 
 
+def _compute_longitude(longitude, positive_longitude, wrap_angle):
+    if positive_longitude == "east":
+        orient = 1
+    elif positive_longitude == "west":
+        orient = -1
+    return Longitude(
+        orient * longitude, u.deg, wrap_angle=wrap_angle * u.deg, copy=False
```

Comment:
This way you always make a copy of the longitude, which costs extra time and memory (see main comments).

Reply:
Also, `wrap_angle` should be a `Quantity` already!